### DISCUSSION TITLE
Use GPU for reranker

### DISCUSSION THREAD

**Discussion by SD-fan** [open-webui/open-webui#3977] [Q&A]
I serve the LLM and embedding model via Ollama - both running on GPUs.
Unfortunately, setting a reranker in OI runs the reranking model always on CPU - although OI is running on docker with --gpus=all.
How can I run the reranker on GPU?

**Comment by justinh-rahb** [comment]
`ghcr.io/open-webui/open-webui:cuda`

**Reply by SD-fan** [reply]
@justinh-rahb  I run ghcr.io/open-webui/open-webui:ollama. 
As Ollama runs on GPU in the container I'd expect reranker to support GPU too. 
Installing Ollama separately to get reranker GPU with the :cuda image is not a convenient solution.

**Reply by justinh-rahb** [reply]
![best_i_can_do_2_containers](https://github.com/user-attachments/assets/1afca3c4-2ad5-45ee-8e2d-e83d31cfcdc3)

**Reply by SD-fan** [reply]
Will do. Thx.

**Reply by SD-fan** [reply]
For the future it would be great to add support for reranker API like https://api.jina.ai/redoc#tag/rerank
Thanks again for your support! @justinh-rahb 🙏

**Reply by gilbrotheraway** [reply]
> @justinh-rahb I run ghcr.io/open-webui/open-webui:ollama. As Ollama runs on GPU in the container I'd expect reranker to support GPU too. Installing Ollama separately to get reranker GPU with the :cuda image is not a convenient solution.

maybe you're downloading the safetensors version? try a gguf and see if ollama catches up?

**Reply by SD-fan** [reply]
> > @justinh-rahb I run ghcr.io/open-webui/open-webui:ollama. As Ollama runs on GPU in the container I'd expect reranker to support GPU too. Installing Ollama separately to get reranker GPU with the :cuda image is not a convenient solution.
> 
> maybe you're downloading the safetensors version? try a gguf and see if ollama catches up?

The solution with two containers works fine with GGUF for almost a year now. Problem solved since. 🤷

**Reply by rgaricano** [reply]
Right now reranker isn't outsourced, it run throught open-webui not ollama.

**Reply by galvanoid** [reply]
Can anyone help me with this?

I installed the cuda version of owui (ghcr.io/open-webui/open-webui:cuda), and when I activate reranker, it still uses the CPU (only for reranker; the rest of the models use GPUs).
Does the :cuda version have the same connection to the ollama API? (localhost:11434 in my case)

Thanks.
